src/main.rs

```diff
@@ -160,6 +173,68 @@ impl TestConfig {
 		}
 	}
 
+	fn run<A: memory::AddressSpace>(
+		&self,
+		cpu_state: &mut cpu::State<A>,
+		rom_path: &str,
+	) -> TestResult {
+		self.configure(cpu_state);
+
+		// Push the return address 0xFFFF onto the stack.
+		// If pc == 0xFFFF the test is complete.
+		// TODO: make the success address configurable.
+		cpu_state.write(cpu_state.sp - 1, 0xFF);
+		cpu_state.write(cpu_state.sp - 2, 0xFF);
+		cpu_state.sp -= 2;
+
+		let condition = loop {
+			match cpu_state.tick() {
+				cpu::TickResult::Ok => {}
+				cpu::TickResult::Halt => break Ok(()),
+				cpu::TickResult::Stop => break Ok(()),
+				cpu::TickResult::Break => {
+					if self.enable_breakpoints {
+						println!("{rom_path}: BREAKPOINT in {} \n{cpu_state}", self.name);
+					}
+				}
+				cpu::TickResult::Debug => {
+					if self.enable_breakpoints {
+						println!("{rom_path}: DEBUG in {}\n{cpu_state}", self.name);
+					}
+				}
```

Comment:
This probably belongs in some "logger" abstraction. `rom_path` being passed in is an uncomfortable hack.